### PR TITLE
DBW Node - First Commit

### PR DESCRIPTION
**Pull request by dehavenm**
If anyone wants to try tuning the PID gain, it's in lines 24-31 of twist_controller.py

### CODE REVIEW

**Review by olala7846 — APPROVED** [review]
LTGM 😄 great job team

File: ros/src/twist_controller/twist_controller.py [hunk 1]
```diff
@@ -2,13 +2,64 @@
...
+        
+
+
+    def control(self, velocity_cmd, current_velocity, angular_velocity_cmd, dbw_enabled):
+        
+        if dbw_enabled:
+            steer = self.yaw_controller.get_steering(velocity_cmd, angular_velocity_cmd, current_velocity)
+            if velocity_cmd > current_velocity:
+                error = velocity_cmd - current_velocity
+                throttle = self.pid_controller.step(error, 0.02)
```
Comment (abroekhof) [inline ros/src/twist_controller/twist_controller.py]: rate.sleep() is not guaranteed to produce 50hz, so we might need to calculate the step time from the loop above

**Review by abroekhof — APPROVED** [review]
Looks good

File: ros/src/twist_controller/dbw_node.py [hunk 2]
```diff
@@ -45,33 +45,54 @@ def __init__(self):
...
+                                    
 
         # TODO: Create `TwistController` object
-        # self.controller = TwistController(<Arguments you wish to provide>)
+        self.controller = TwistController(vehicle_mass, brake_deadband, decel_limit, accel_limit, wheel_radius, wheel_base, steer_ratio, max_lat_accel, max_steer_angle)
 
         # TODO: Subscribe to all the topics you need to
-
+        self.current_velocity_sub = rospy.Subscriber("/vehicle/current_velocity", TwistStamped, self.current_velocity_callback)
+        self.twist_cmd_sub = rospy.Subscriber("/vehicle/twist_cmd", TwistStamped, self.twist_cmd_callback)
```
Comment (kjames88) [inline ros/src/twist_controller/dbw_node.py:64]: I think this the wrong subscription (/vehicle/twist_cmd).  From the log at the top of dbw_node.py:

"You will subscribe to `/twist_cmd` message which provides the proposed linear and angular velocities."

File: ros/src/twist_controller/twist_controller.py [hunk 3]
```diff
@@ -2,13 +2,64 @@
...
+        if dbw_enabled:
+            steer = self.yaw_controller.get_steering(velocity_cmd, angular_velocity_cmd, current_velocity)
+            if velocity_cmd > current_velocity:
+                error = velocity_cmd - current_velocity
+                throttle = self.pid_controller.step(error, dt)
+                brake = 0.0
+                self.brake_controller.reset()
+            else:
+                error = current_velocity - velocity_cmd
+                brake = brake.pid_controller.step(error, dt)
```
Comment (joshchao39) [inline ros/src/twist_controller/twist_controller.py:55]: I guess `brake.pid_controller` here should be `self.brake_controller`

**Pull request closed without merging**